xmipptomo/protocols/protocol_detect_landmarks_TS.py

```diff
@@ -0,0 +1,234 @@
+            tsObjId = ts.getObjId()
+            cisID = self._insertFunctionStep(self.convertInputStep,
+                                             tsObjId,
+                                             prerequisites=[])
+            dlsID = self._insertFunctionStep(self.detectLandmarksStep,
+                                             tsObjId,
+                                             prerequisites=[cisID])
+            cosID = self._insertFunctionStep(self.createOutputStep,
+                                             tsObjId,
+                                             prerequisites=[dlsID])
+            allcosId.append(cosID)
+
+        self._insertFunctionStep(self.closeOutputSetsStep,
+                                 prerequisites=allcosId)
+
+    # --------------------------- STEPS functions ----------------------------
+    def convertInputStep(self, tsObjId):
+        ts = self.inputSetOfTiltSeries.get()[tsObjId]
+        tsId = ts.getTsId()
+
+        extraPrefix = self._getExtraPath(tsId)
+        tmpPrefix = self._getTmpPath(tsId)
+
+        path.makePath(tmpPrefix)
+        path.makePath(extraPrefix)
+
+        firstItem = ts.getFirstItem()
+
+        """Apply the transformation form the input tilt-series"""
+        # Use Xmipp interpolation via Scipion
+        if firstItem.hasTransform():
+            avgRotAngle = utils.calculateAverageRotationAngleFromTM(ts)
+            swap = True if (avgRotAngle > 45 or avgRotAngle < -45) else False
+
+            outputTsFileName = os.path.join(tmpPrefix, firstItem.parseFileName())
+            ts.applyTransform(outputTsFileName, swapXY=swap)
+
+        else:
+            outputTsFileName = os.path.join(tmpPrefix, firstItem.parseFileName())
+            ts.applyTransform(outputTsFileName)
+
+    def detectLandmarksStep(self, tsObjId):
+        ts = self.inputSetOfTiltSeries.get()[tsObjId]
+        tsId = ts.getTsId()
+
+        extraPrefix = self._getExtraPath(tsId)
+        tmpPrefix = self._getTmpPath(tsId)
+
+        firstItem = ts.getFirstItem()
+
+        params = {
+            'i': os.path.join(tmpPrefix, firstItem.parseFileName() + ":mrcs"),
+            'o': os.path.join(extraPrefix, OUTPUT_COORDS_FILENAME),
+            'thrSD': self.thrSD.get(),
+            'samplingRate': self.inputSetOfTiltSeries.get().getSamplingRate(),
+            'fiducialSize': self.fiducialSize.get() * 10,
+            'targetLMsize': self.targetLMsize.get(),
+            'numberFTdirOfDirections': self.numberFTdirOfDirections.get()
+        }
+
+        args = "-i %(i)s " \
+               "-o %(o)s " \
+               "--thrSD %(thrSD).2f " \
+               "--samplingRate %(samplingRate).2f " \
+               "--fiducialSize %(fiducialSize).2f " \
+               "--targetLMsize %(targetLMsize).4f " \
+               "--numberFTdirOfDirections %(numberFTdirOfDirections)d "
+
+        self.runJob('xmipp_tomo_detect_landmarks', args % params)
+
+    def createOutputStep(self, tsObjId):
```

Comment:
do you tolerate tilt series without landmarks in your output?

Reply:
Fixes, in no landmark is detected in tilt series, landmark model is no added to the output.